app/plugins/shards/shards.js

```diff
@@ -0,0 +1,238 @@
+      'format(\'%s.%s.%s\', schema_name, table_name, partition_ident) AS key, ' +
+      '_node[\'id\'] AS node_id, _node AS nodeInfo, state, routing_state, relocating_node, primary ' +
+      'FROM sys.shards ';
+
+    var cols = ['id', 'size', 'min_lucene_version', 'num_docs', 'orphan_partition', 'table_name', 'schema_name', 'partition_ident', 'fqn', 'key', 'node_id', 'nodeInfo', 'state', 'routing_state', 'relocating_node', 'primary'];
+
+    ShardService.execute = function () {
+      var deferred = $q.defer(),
+        promise = deferred.promise;
+      SQLQuery.execute(stmt, {}, false, false, true)
+        .success(function (query) {
+          var result = queryResultToObjects(query, cols);
+          deferred.resolve(result);
+        })
+        .error(function () {
+          deferred.reject();
+        });
+
+      return promise;
+    };
+
+    return ShardService;
+  })
+  .factory('ShardsIntervalService', function (ShardService, $timeout, $q, $rootScope) {
+    var shardsIntervalService = {};
+
+    var retryCount = 0;
+    var poll = function (force) {
+      $q.when(ShardService.execute())
+        .then(function (response) {
+          shardsIntervalService.response = response;
+          $rootScope.$broadcast('shard-query-done');
+          retryCount = 0;
+          if (force !== true) {
+            $timeout(poll, 60000);
+          }
+        }).catch(function () {
+          shardsIntervalService.response = [];
+          $rootScope.$broadcast('shard-query-done');
+          retryCount++;
+          if (force !== true) {
+            $timeout(poll, 60000 * retryCount);
+          }
+        });
+    };
+    // Initial poll
+    $timeout(poll, 2000);
+    shardsIntervalService.refresh = function () {
+      poll(true);
+    };
+    return shardsIntervalService;
+  })
+  .filter('shardStateClass', function () {
+    return function (state, primary) {
+      var cssClass = '';
+
+      if (primary === true) {
+        cssClass = 'cr-shard-primary ';
+      }
+
+      switch (state) {
+      case 'CREATED':
+        return cssClass + 'cr-shard-state--green';
+      case 'STARTED':
+        return cssClass + 'cr-shard-state--green';
+      case 'RELOCATED':
+        return cssClass + 'cr-shard-state--green';
+      case 'INITIALIZING':
+        return cssClass + 'cr-shard-state--yellow';
+      case 'POST_RECOVERY':
+        return cssClass + 'cr-shard-state--yellow';
+      case 'RECOVERING':
+        return cssClass + 'cr-shard-state--red';
+      case 'UNASSIGNED':
+        return cssClass + 'cr-shard-state--grey';
+      default:
+        return cssClass + 'cr-shard-state--grey';
+      }
+    };
+  })
+  .controller('ShardsController', function ($scope, $q, ShardsIntervalService, $location, $filter, $rootScope) {
+    $scope.response = [] || ShardsIntervalService.response;
```

Comment:
not sure what this OR operator should do in this case.
[] is a static empty array and does never change.
maybe you wanted to flip the right and left side of the operation?